### PR TITLE
Fix Heavy path decoy duplicate detection to use modified sequences

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Summary

Fixes a bug in `generateDecoys()` (Heavy path) where the duplicate detection check used unmodified sequences while the lookup map contained modified sequences, causing the check to almost never match.

**The bug:**
- Line 542: Map populated with `getModifiedPeptideSequence_(peptide)` → e.g., `"AC(UniMod:4)DEK2"`
- Line 603: Check used `peptide.sequence` → e.g., `"KEDCA2"` (unmodified, reversed)
- These formats don't match, so duplicates weren't detected

**The fix:**
- Line 604 now uses `getModifiedPeptideSequence_(peptide)` consistently

This aligns the Heavy path with the Light path behavior, which correctly uses modified sequences for duplicate detection.

## Impact

- Eliminates ~0.5% discrepancy between Heavy and Light path decoy counts
- Correctly detects when a reversed decoy (with modifications) matches a target sequence
- No change to output for datasets without this edge case

## Test plan

- [x] `MRMDecoy_test` passes (includes `generateDecoysLight_modified_sequence_duplicate_detection`)
- [x] `MRMAssay_test` passes
- [x] `OpenSwathDecoyGenerator` TOPP tests pass (20 tests)
- [x] `OpenSwathAssayGenerator` TOPP tests pass (12 tests)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected duplicate-decoy detection so modified peptide sequences (including modifications) are considered, avoiding false duplicate classification.

* **Tests**
  * Added tests validating decoy generation for peptides with identical base sequences but different modifications, ensuring distinct decoys and preserved transitions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by timosachsenberg** [comment]
Maybe give this a quick sanity check? Because it could be that the original (heavy) path had a small bug and produced too many decoys (like <0.5%)

**Comment by jcharkow** [comment]
This seems to look reasonable on my end. It looks like there might have been a bug with the heavy implementation. Before merging would like for @singjc to also have a look. 

This seems to fix the discrepancy in the number of decoys that I was experiencing, now we see that the decoy peptides are both 165445

File: src/tests/class_tests/openms/source/MRMDecoy_test.cpp [hunk 1]
```diff
@@ -1135,6 +1135,150 @@ START_SECTION([EXTRA] generateDecoysLight_modified_sequence_duplicate_detection)
...
+  transition2.setPeptideRef("peptide2");
+  transition2.setProductMZ(501.0);
+  transition2.setPrecursorMZ(601.0);
+  transition2.setDetectingTransition(true);
+
+  CVTerm cv_term2;
+  cv_term2.setCVIdentifierRef("MS");
+  cv_term2.setAccession("MS:1001220");
+  cv_term2.setName("frag: y ion");
+  transition2.addProductCVTerm(cv_term2);
```
Comment (jcharkow) [inline src/tests/class_tests/openms/source/MRMDecoy_test.cpp:1220]: I don't think this is how it should be encoded however not sure if this matters for the test

**Review by jcharkow — APPROVED** [review]
Minor comment but good to merge on my end.

File: src/tests/class_tests/openms/source/MRMDecoy_test.cpp (hunk 1 above)
Comment (timosachsenberg) [reply src/tests/class_tests/openms/source/MRMDecoy_test.cpp:1220]: thanks for checking. hmm I like it when tests also show how to use things. do you have a suggestion / prompt to update this? I can add it.

Comment (jcharkow) [reply src/tests/class_tests/openms/source/MRMDecoy_test.cpp:1220]: You can tell it to look at the `createTransition_` method in `TransitionTSVFile.cpp`  

Two main things. 
1. Fragment ions are not annotated directly but rather under the `TargetedExperiment::Interpretation object. 
e.g. set in interpretation.iontype
3. CV terms for y ions are not set directly with strings but rather use an enum `TargetedExperiment::IonType::YIon`

Comment (jpfeuffer) [reply src/tests/class_tests/openms/source/MRMDecoy_test.cpp:1220]: It would actually be nice to reuse those enums and interpretation types generally in annotated spectra. Maybe they are not in the best place if no one found them so far.

Comment (timosachsenberg) [reply src/tests/class_tests/openms/source/MRMDecoy_test.cpp:1220]: No strong opinion but there are standardized string encodings that would  require less space (e.g. typically one char per ion type). I would probably go with what is (now) standardized and try to make this a bit more consistent in OpenMS

Comment (jpfeuffer) [reply src/tests/class_tests/openms/source/MRMDecoy_test.cpp:1220]: Yes can also be the other way round, as long as it is consistent.
